src/computability/timed/lemmas.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2022 Tomaz Gomes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tomaz Gomes
+-/
+import data.nat.log
+import tactic.linarith
+/-!
+# lemmas
+  This file defines and proves some lemmas about the logarithm function (with base set fixed to 2)
+  and other arithmetic relations that will be used to prove merge sort's time complexity.
+## Main Results
+  - log_pred : ∀ (a : ℕ) , nat.log 2 a - 1 = nat.log 2 (a / 2)
+  - log_2_times : ∀ (a : ℕ), 2 * nat.log 2 (a + 2) ≤ a + 2
+-/
+
+lemma log_pred : ∀ (a : ℕ) , nat.log 2 a - 1 = nat.log 2 (a / 2)
```

Comment:
I think this is a lot cleaner by translating to `+` and using `nat.log_of_one_lt_of_le`. (There may be ways to make this slicker, this was just the first I came up with.)

```lean
lemma log_pred (n : ℕ) : nat.log 2 n - 1 = nat.log 2 (n / 2) :=
if h : n < 2 then by simp [nat.log_of_lt h] else
(nat.sub_eq_iff_eq_add (by rw ← nat.pow_le_iff_le_log; linarith)).mpr 
  (nat.log_of_one_lt_of_le (by norm_num) (le_of_not_lt h))
```